classes/statistics/PKPTemporaryItemRequestsDAO.php

```diff
@@ -70,16 +80,42 @@ public function deleteByLoadId(string $loadId): void
      *
      * See https://www.projectcounter.org/code-of-practice-five-sections/7-processing-rules-underlying-counter-reporting-data/#counting
      */
-    public function compileUniqueClicks(): void
+    public function compileUniqueClicks(string $loadId): void
     {
         if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
-            DB::statement("DELETE FROM {$this->table} usur WHERE EXISTS (SELECT * FROM (SELECT 1 FROM {$this->table} usurt WHERE usurt.load_id = usur.load_id AND usurt.ip = usur.ip AND usurt.user_agent = usur.user_agent AND usurt.context_id = usur.context_id AND usurt.submission_id = usur.submission_id AND EXTRACT(HOUR FROM usurt.date) = EXTRACT(HOUR FROM usur.date) AND usur.line_number < usurt.line_number) AS tmp)");
+            DB::statement(
```

Comment:
I didn't inspect the query, but isn't it possible to write it using the Laravel stuff to have a single query (and just add exceptions for the `EXTRACT`/`TIMESTAMPDIFF`)?

Reply:
@jonasraoni, PostgreSQL does not support inner joint with delete, and the other way around too (MySQL does not understand the PostgreSQL's query here), so I would leave this as is for now, OK?

Reply:
For the query `DB::table('submissions', 'a')->join('submissions AS b', 'b.submission_id', '=', 'a.submission_id')->delete();`, Laravel outputs this for PostgreSQL:

```sql
delete from "submissions" as "a" where "ctid" in (
	select "a"."ctid" from "submissions" as "a" inner join "submissions" as "b" on "b"."submission_id" = "a"."submission_id"
)
```

I think it will work fine for your case, but I don't think this is a blocker 😁

Reply:
OK, let me try it with PostgreSQL... I am a little bit afraid to change anything... but I will test it thoroughly... :sweat_smile:

Reply:
I created a new issue for that -- to test and see if we could use the Laravel query builder -- in order to finish this issue as soon as possible and provide affected users with the patch. S. https://github.com/pkp/pkp-lib/issues/9636.

Reply:
Not a big deal, I marked this as resolved 😁